### PR TITLE
ITF-01-03 - Init chat project

### PR DESCRIPTION
**Pull request by yevhenip**
Init chat project

### CODE REVIEW

File: server/Services/Chat/Chat.Hub/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿{
+  "iisSettings": {
```
Comment (neistow) [inline server/Services/Chat/Chat.Hub/Properties/launchSettings.json:2]: Remove iis settings

File: server/Services/Chat/Chat.Hub/dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
+WORKDIR /build
+
+COPY ["Services/Chat/Chat.Hub/Chat.Hub.csproj", "Services/Chat/Chat.Hub/Chat.Hub.csproj"]
```
Comment (neistow) [inline server/Services/Chat/Chat.Hub/dockerfile:4]: Check dockerfile in ITF.Api. You're missing all other projects

File: server/Services/Chat/Chat.Hub/dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+RUN dotnet restore "Services/Chat/Chat.Hub/Chat.Hub.csproj"
+
+COPY . .
+WORKDIR "/build/Services/Chat/Chat.Hub"
+
+RUN dotnet build "Chat.Hub.csproj" -c Release --no-restore
+RUN dotnet publish "Chat.Hub.csproj" -c Release -o /published-app --no-restore
+
+FROM mcr.microsoft.com/dotnet/aspnet:6.0
+WORKDIR /identity-server
```
Comment (neistow) [inline server/Services/Chat/Chat.Hub/dockerfile:15]: Rename copypaste

File: server/docker-compose.local.yml [hunk 4]
```diff
@@ -24,6 +24,17 @@ services:
...
       - ~/.aspnet/https:/https:ro
+  chat:
+    restart: on-failure
+    ports:
+      - "3001:3001"
+    environment:
+      ASPNETCORE_URLS: "https://+:3001"
+      ASPNETCORE_Kestrel__Certificates__Default__Password: 12345
+      ASPNETCORE_Kestrel__Certificates__Default__Path: /https/aspnetapp.pfx
+      ConnectionStrings__Postgres: "Host=database;Username=postgres;Password=postgres;Database=chat"
```
Comment (neistow) [inline server/docker-compose.local.yml:35]: We'll have single database, but with different schemas

File: server/Services/Chat/Chat.Infrastructure/ApplicationDbContext.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+using Microsoft.EntityFrameworkCore;
+
+namespace Chat.Infrastructure;
+
+public class ApplicationDbContext : DbContext
+{
+    public DbSet<ChatRoom> Chats { get; set; }
+    public DbSet<Message> Messages { get; set; }
+
+    public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
```
Comment (neistow) [inline server/Services/Chat/Chat.Infrastructure/ApplicationDbContext.cs:11]: Configure schema

File: server/docker-compose.yml [hunk 6]
```diff
@@ -13,5 +13,11 @@ services:
       dockerfile: ./Services/ITF/ITF.Api/dockerfile
     depends_on:
       - identity_server
+  chat:
+    build:
+      context: .
+      dockerfile: ./Services/Chat/Chat.Hub/dockerfile
+    depends_on:
+      - database
```
Comment (neistow) [inline server/docker-compose.yml:21]: For now - depens on identity_server

File: server/Services/Chat/Chat.Domain/ChatRoom.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Chat.Domain;
+
+public class ChatRoom
+{
+    public Guid Id { get; set; }
+    public Guid RecruiterId { get; set; }
+    public Guid ApplicantId { get; set; }
```
Comment (neistow) [inline server/Services/Chat/Chat.Domain/ChatRoom.cs:7]: Note: domain models were changed. See ITF.Domain/Entities.

Employer -> Recruiter
Applicant -> Developer

Comment (neistow) [reply server/Services/Chat/Chat.Domain/ChatRoom.cs:7]: Also... Will it be a good idea to tie chat implementation to recruiters/developers?

File: server/Services/Chat/Chat.Hub/Properties/launchSettings.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      "applicationUrl": "http://localhost:44062",
+      "sslPort": 44366
+    }
+  },
+  "profiles": {
+    "Chat.Hub": {
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "https://localhost:7249;http://localhost:5249",
```
Comment (neistow) [inline server/Services/Chat/Chat.Hub/Properties/launchSettings.json:15]: Set applicationUrl to https://localhost:3001

File: server/Services/Chat/Chat.Hub/Properties/launchSettings.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    "iisExpress": {
+      "applicationUrl": "http://localhost:44062",
+      "sslPort": 44366
+    }
+  },
+  "profiles": {
+    "Chat.Hub": {
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
```
Comment (neistow) [inline server/Services/Chat/Chat.Hub/Properties/launchSettings.json:14]: Remove launch browser

**Review by neistow — CHANGES_REQUESTED** [review]

File: server/Services/Chat/Chat.Domain/ChatRoom.cs (hunk 7 above)
Comment (yevhenip) [reply server/Services/Chat/Chat.Domain/ChatRoom.cs:7]: Will chat be used for recruiters/recruiters or developers/developers?

**Pull request merged**